bin/datatool/src/util.rs

```diff
@@ -0,0 +1,417 @@
+//! Utility functions for Strata `datatool` binary.
+//!
+//! It contains functions for generating keys, parsing amounts, and constructing
+//! network parameters.
+//! These functions are called from the CLI's subcommands.
+use std::{
+    fs,
+    path::{Path, PathBuf},
+};
+
+use bitcoin::{
+    base58,
+    bip32::{Xpriv, Xpub},
+    Network,
+};
+use rand::{CryptoRng, RngCore};
+use strata_key_derivation::{
+    operator::{convert_base_xpub_to_message_xpub, convert_base_xpub_to_wallet_xpub, OperatorKeys},
+    sequencer::SequencerKeys,
+};
+use strata_primitives::{
+    block_credential,
+    buf::Buf32,
+    keys::ZeroizableXpriv,
+    operator::OperatorPubkeys,
+    params::{ProofPublishMode, RollupParams},
+    vk::RollupVerifyingKey,
+};
+use strata_sp1_guest_builder::GUEST_CHECKPOINT_VK_HASH_STR;
+
+use crate::args::{
+    CmdContext, SubcOpXpub, SubcParams, SubcSeqPrivkey, SubcSeqPubkey, SubcXpriv, Subcommand,
+};
+
+/// Sequencer key environment variable.
+const SEQKEY_ENVVAR: &str = "STRATA_SEQ_KEY";
+
+/// Operator key environment variable.
+const OPKEY_ENVVAR: &str = "STRATA_OP_KEY";
+
+/// The default network to use.
+///
+/// Right now this is [`Network::Signet`].
+const DEFAULT_NETWORK: Network = Network::Signet;
+
+/// Resolves a [`Network`] from a string.
+pub(super) fn resolve_network(arg: Option<&str>) -> anyhow::Result<Network> {
+    match arg {
+        Some("signet") => Ok(Network::Signet),
+        Some("regtest") => Ok(Network::Regtest),
+        Some(n) => anyhow::bail!("unsupported network option: {n}"),
+        None => Ok(DEFAULT_NETWORK),
+    }
+}
+
+/// Executes a `gen*` subcommand.
+pub(super) fn exec_subc(cmd: Subcommand, ctx: &mut CmdContext) -> anyhow::Result<()> {
+    match cmd {
+        Subcommand::Xpriv(subc) => exec_genxpriv(subc, ctx),
+        Subcommand::SeqPubkey(subc) => exec_genseqpubkey(subc, ctx),
+        Subcommand::SeqPrivkey(subc) => exec_genseqprivkey(subc, ctx),
+        Subcommand::OpXpub(subc) => exec_genopxpub(subc, ctx),
+        Subcommand::Params(subc) => exec_genparams(subc, ctx),
+    }
+}
+
+/// Executes the `genxpriv` subcommand.
+///
+/// Generates a new [`Xpriv`] that will [`Zeroize`](zeroize) on [`Drop`] and writes it to a file.
+fn exec_genxpriv(cmd: SubcXpriv, ctx: &mut CmdContext) -> anyhow::Result<()> {
+    if cmd.path.exists() && !cmd.force {
+        anyhow::bail!("not overwriting file, add --force to overwrite");
+    }
+
+    let xpriv = gen_priv(&mut ctx.rng, ctx.bitcoin_network);
+    let buf = xpriv.encode();
+    let s = base58::encode_check(&buf);
```

Comment:
Both of these are secret values that are not zeroized.

Reply:
Done in 965aaee1b7b4318eb8a72b0a873ab5592b15bde6